From e9dba441872ed8c4349ed4b6c0256e23decf8080 Mon Sep 17 00:00:00 2001
From: b461 <b461@borqs.com>
Date: Thu, 23 Aug 2012 13:35:44 +0800
Subject: [PATCH 03/18] [PORT FROM MAIN] [PORT FROM R3]FM support in IHF
 (Borqs)

BZ: 56091

Contribution from Borqs
See detail below

fmradio: Integration of Borqs contribution (Date: Sep 4, 2012).

BZ: 55486

Original commit message:

Sun Liang: enable fm radio through speaker

Change-Id: I803f53c1261e75598713a9c3d40420503cc11ede
Orig-Change-Id: I45380b13b866fc3e953606cd1dec8b7d154a25e0
Orig-Change-Id: I776fc368537ff82642c5905380161365ff758b99
borqsbtx:0034955
rootcause: add set_fm_mode method in audio policy hal interface
Reviewed-on: https://bkbgerrit.borqs.com/gerrit/7870
Tested-by: svnadmin <svnadmin@borqs.com>
Reviewed-by: b169 <b169@borqs.com>
Reviewed-on: http://android.intel.com:8080/75588
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 include/hardware/audio_policy.h |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/include/hardware/audio_policy.h b/include/hardware/audio_policy.h
index 9e145f8..dde98b8 100644
--- a/include/hardware/audio_policy.h
+++ b/include/hardware/audio_policy.h
@@ -105,6 +105,9 @@ struct audio_policy {
     void (*set_ringer_mode)(struct audio_policy *pol, uint32_t mode,
                             uint32_t mask);
 
+    /* indicate a change in fm radio mode */
+    void (*set_fm_mode)(struct audio_policy *pol, uint32_t mode);
+
     /* force using a specific device category for the specified usage */
     void (*set_force_use)(struct audio_policy *pol,
                           audio_policy_force_use_t usage,
-- 
1.7.9

